### PR TITLE
[WIP] cyrus_imapd: init at 3.0.11

### PR DESCRIPTION
**Pull request by ghost**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by petabyteboy** [comment]
#45012

**Comment by petabyteboy** [comment]
#54012

**Comment by petabyteboy** [comment]
The perl scripts can not find the required modules, but otherwise it looks ok.

**Comment by petabyteboy** [comment]
Can someone tell me where the bad indent is? OfBorg says it should be in line 1, but it looks fine to me.

File: pkgs/servers/mail/cyrus-imapd/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "cyrus-imapd";
+  version = "3.0.10";
+
+  src = fetchurl {
+    url = "https://www.cyrusimap.org/releases/${pname}-${version}.tar.gz";
+    sha256 = "02q2j4qkj58gz4kjs0s19j6p1d6nhdh5pzrhkilgigny7pglpcm2";
+  };
+
+  nativeBuildInputs = [ pkgconfig perl ];
+  buildInputs = [ bison flex openssl jansson cyrus_sasl icu ];
```
Comment (aanderse) [inline pkgs/servers/mail/cyrus-imapd/default.nix]: I haven't looked at the code for this project, but usually `bison` and `flex` are `nativeBuildInputs`. That isn't the case here?

**Review by aanderse — COMMENTED** [review]
As always looking good @petabyteboy :+1:

File: pkgs/servers/mail/cyrus-imapd/default.nix (hunk 1 above)
Comment (petabyteboy) [reply pkgs/servers/mail/cyrus-imapd/default.nix]: You're right, thanks ^^

**Comment by FRidh** [comment]
@grahamcofborg eval

**Comment by teto** [comment]
@petabyteboy why the WIP ? because of the perl issue : it can't find perl belonging to cyrus_imap or 3rd party plugins ? you might need to create an environment perl.withPackages ?
I am loosely interested in the software (mainly because of the JMAP implementation), I wonder if you plan to add a module (which could make testing the software easier)

**Comment by petabyteboy** [comment]
> @petabyteboy why the WIP ? because of the perl issue : it can't find perl belonging to cyrus_imap or 3rd party plugins ? you might need to create an environment perl.withPackages ?
> I am loosely interested in the software (mainly because of the JMAP implementation), I wonder if you plan to add a module (which could make testing the software easier)

Yes because of the perl thing. 
When I start sieveshell, I get this:
```
Can't load '/nix/store/qv1s3nfwqdk7kncckv0mm554dm1qf5vz-cyrus-imapd-3.0.10/lib/perl5/site_perl/5.28.2/x86_64-linux-thread-multi/auto/Cyrus/SIEVE/managesieve/managesieve.so' for module Cyrus::SIEVE::managesieve: /nix/store/qv1s3nfwqdk7kncckv0mm554dm1qf5vz-cyrus-imapd-3.0.10/lib/perl5/site_perl/5.28.2/x86_64-linux-thread-multi/auto/Cyrus/SIEVE/managesieve/managesieve.so: undefined symbol: sasl_client_step at /nix/store/xnr1miqcjqsvqb6n23zavrvnnyax49bi-perl-5.28.2/lib/perl5/5.28.2/x86_64-linux-thread-multi/DynaLoader.pm line 193.
```

**Comment by petabyteboy** [comment]
I'm wondering if perl should be in nativeBuildInputs or in buildInputs. The perl interpreter is used during the build, but it is also needed at run-time.

**Comment by petabyteboy** [comment]
I have fixed PERL5LIB and sieveshell can find the perl libraries in the output, but I am now stuck with a different problem I can not solve:

```
pbb@amalthea ~/p/nixpkgs> result/bin/sieveshell
Can't load '/nix/store/4g31sxc451qc2425a62cks7r0pvl0p4h-cyrus-imapd-3.0.12/lib/perl5/site_perl/5.30.1/x86_64-linux-thread-multi/auto/Cyrus/SIEVE/managesieve/managesieve.so' for module Cyrus::SIEVE::managesieve: /nix/store/4g31sxc451qc2425a62cks7r0pvl0p4h-cyrus-imapd-3.0.12/lib/perl5/site_perl/5.30.1/x86_64-linux-thread-multi/auto/Cyrus/SIEVE/managesieve/managesieve.so: undefined symbol: sasl_client_step at /nix/store/9a8q20iyq9si8f9sxm2afpd57gap4qlj-perl-5.30.1/lib/perl5/5.30.1/x86_64-linux-thread-multi/DynaLoader.pm line 193.
 at /nix/store/4g31sxc451qc2425a62cks7r0pvl0p4h-cyrus-imapd-3.0.12/bin/.sieveshell-wrapped line 65.
Compilation failed in require at /nix/store/4g31sxc451qc2425a62cks7r0pvl0p4h-cyrus-imapd-3.0.12/bin/.sieveshell-wrapped line 65.
BEGIN failed--compilation aborted at /nix/store/4g31sxc451qc2425a62cks7r0pvl0p4h-cyrus-imapd-3.0.12/bin/.sieveshell-wrapped line 65.
pbb@amalthea ~/p/nixpkgs>
```

**Comment by petabyteboy** [comment]
Small update: I am still interested but could not find a solution to the last problem yet. Suggestions are welcome.

**Comment by aanderse** [comment]
@petabyteboy I bet @volth knows the solution to this problem without having to do any research at all...

**Comment by volth** [comment]
I have no immediate solution :(

**Comment by petabyteboy** [comment]
I'm not planning to work on this anymore

**Pull request closed without merging**